### PR TITLE
fix: [CDS-53876]: added support for updating issue type in Jira Update Step

### PR DESCRIPTION
**Pull request by naman-goenka**
## Describe your changes
- Supported "issuetype" field in Jira
- example Jira update step yaml
```
              - step:
                  type: JiraUpdate
                  name: Jira Update_1
                  identifier: JiraUpdate_1
                  spec:
                    connectorRef: account.DoNotDelete_JiraConnectorForAutomationDoNotDelete
                    issueKey: TJI-106767
                    transitionTo:
                      transitionName: ""
                      status: To Do
                    fields:
                      - name: 1-line Update
                        value: "n"
                      - name: Issue Type
                        value: Bug
```
- Removed Issue Type field from appearing in createMetadata both in Server (new and old), Cloud 
- Added actual API response in coverage to validate the above.
- Testing: issue type is changed both in Server and cloud, updateMetadata call now returns issue type as well for UI to populate dropdown,  createMetadata doesn't include issuetype field. Also tested backward compatibility with older delegates. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44307)

### CODE REVIEW

**Comment by naman-goenka** [comment]
trigger messagecheck

**Comment by naman-goenka** [comment]
trigger codeformat

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44307)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Comment by naman-goenka** [comment]
trigger codeformat

File: 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueUtilsNG.java [hunk 1]
```diff
@@ -257,4 +259,16 @@ private Object convertIssueLinkToFinalValue(JiraFieldNG field, String name, Stri
     issueLinkMap.put("key", value);
     return issueLinkMap;
   }
+
+  private Object convertIssueTypeToFinalValue(JiraFieldNG field, String name, String value) {
+    // reference
+    // https://community.atlassian.com/t5/Jira-questions/How-can-I-change-an-issuetype-on-an-existing-issue-via-the-REST/qaq-p/801750#:~:text=Then%20I%20executed%20PUT%20%C2%A0/rest/api/2/issue/issuekey%20with%20the%20following%20JSON%3A
```
Comment (sribalij) [inline 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueUtilsNG.java:265]: Do we need this url? Can we may be add comment instead if we need it?

**Review by sribalij — APPROVED** [review]

**Review by Y0709 — APPROVED** [review]
> Reviewed 4 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44307)* status: :shipit: complete! ✓

**Pull request merged**